### ISSUE TITLE
Launch?

### ISSUE THREAD

**Issue by OliverRadini** [thekroko/uthgard-herald#54]
Has this been 'officially' launched yet? If not what are the blocks to its being launched? It seems to be quite close to a basically usable point, but no progress has been made in quite some time. I could probably spare some time to work on it but I don't want to waste time on useless features.

**Comment by Nerzal** [comment]
I could also lend a helping hand. Just need the issues just needa get a priority so we can fix / implement stuff. The repo seems pretty dead to me atm

**Comment by OliverRadini** [comment]
Probably needs input from someone on the uthgard project now, do they want this to be developed or not?

**Comment by csrenato** [comment]
I can help out as well.

**Comment by burfo** [comment]
Oh look, activity!  I might be able to help too again, a little bit.  @thekroko what's the status on this project?

**Comment by ianrandmckenzie** [comment]
Curious about the status of this as well.

**Comment by Nerzal** [comment]
[08:44] <[KotG]Aragorn> lets shorten this - is there a chance to revive that propject in any form?
[08:44] <@[Uth]Trishin> (possibly because metty wanted something closer to finish than player devs)
[08:45] <@[Uth]Trishin> <[KotG]Aragorn> lets shorten this - is there a chance to revive that propject in any form? <- yes
[08:45] <[KotG]Aragorn> how ?
[08:45] <@[Uth]Trishin> 1. create a frontend with madeup data (you can even make up the data structure that is provided to the mock-up)
[08:46] <@[Uth]Trishin> 2. present that mockup to staff
[08:46] <@[Uth]Trishin> 3. Wait for a dev to build the data API and implement your mockup as a true front 

Sounds like we have to do 3 things:
1. Create a Roadmap which includes what we implement
2. Implement stuff
3. Make metty implement missing api endpoints

**Comment by OliverRadini** [comment]
@Nerzal that's cool, it sounds like a similar story to where we left it. A lot of pages were implemented based on fake data, and we then started to switch some over to real data.

I'm of the opinion that building the front end first is not ideal, and that we'd be wasting time doing that again. Not least because it's been done already and the uthgard devs didn't do much with it. Take a look at the player page - it's massively filled out on the front end, but nothing was ever put in place to back that up with real data. 

To summarise - a lot is already built with fake data. Some is already built with real data. From the lessons of last time this was worked on, this should be built with the api data first. If the uth devs aren't willing to do that then I'd suggest not working on this.

**Comment by Nerzal** [comment]
@Donutttt i understand the frustration. And i'm also of the opinion, that we sould get the API first, before implementing lot's of UIs that never gets filled. As i wish in the other issue (lacking endpoints) i want to gather some data what we have now. My plan is to gather these data on one point and start planning how to go on with this project. We may also be able to convince the uthgard staff to provide api endpoints, without having the ui, if we have a plan how to finish this project.

**Comment by OliverRadini** [comment]
@Nerzal I think you're on the right track. I can maybe put in some time to help out with things, but I think at first we could aim for something much smaller. We should find out what the api endpoints are, and build to those. They would give us a useful enough herald already;, the INC page uses those same endpoints and gives a lot of data.

**Comment by OliverRadini** [comment]
I'd be interested to see whether people are interested in building something much more minimal. I think in the past the problem with this project is that it went for a complicated approach from the start, looking to add too many features.

My suggestion is rather extreme, but I believe it's the best option; start the app code from scratch, use react instead of angular, and aim to release the minimal viable herald that we can devise. Imo angular makes things *way* more longwinded than they need to be, and this can be a rather simple app, if we let it become that. The current code base is hugely overcomplicated; I say that as someone who has worked with it quite extensively. 

If you check the dev branch of my fork of [this repo](https://github.com/Donutttt/uthgard-herald/tree/dev/herald) I have switched it to react already. It was rather easy and I have setup a search page on the index already. 

Any thoughts? I don't think that the current codebase is workable, it has a large amount of redundant code and overcomplicates things that should be easy.
